### PR TITLE
fat: add function type parameter on findRows (close #199)

### LINKED ISSUE #199: Add function type parameter on findRows API
## Version

v3.0.0

## Development Environment

All

## Current Behavior

```js
// before: only using the object (K: column name, V: column value)
grid.findRows({
  artist: 'Birdy',
  price: 10000
});

// after: enable using function type parameter
grid.findRows(function(row) {
   return (/b/ig.test(row.artist) && row.price > 10000);
});
```

### PR DESCRIPTION
**Pull request by seonim-ryu**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)
- [x] It does not introduce a breaking change or has description for the breaking change

### Description

* #199 

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

**Comment by jinwoo-kim-nhn** [comment]
수고하셨습니다. 이견없습니다.

File: src/js/model/data/rowList.js [hunk 1]
```diff
@@ -362,6 +362,25 @@ var RowList = Collection.extend(/** @lends module:model/data/rowList.prototype *
...
+     */
+    findRows: function(conditions) {
+        var rowList = this.toJSON();
+        var foundList;
+
+        if (_.isFunction(conditions)) {
+            foundList = _.filter(rowList, conditions);
+        } else {
+            foundList = _.where(rowList, conditions);
+        }
```
Comment (dongwoo-kim) [inline src/js/model/data/rowList.js]:
```js
if (_.isFunction(conditions)) {
  foundList = this.filter(conditions);
} else {
  foundList = this.where(conditions);
}
return foundList.toJSON();
```
요런 식으로 하는 게 더 효과적이지 않을까 하는 생각이 있습니다 ㅎ

File: test/unit/js/model/data/rowList.simple.spec.js [hunk 2]
```diff
@@ -452,4 +452,83 @@ describe('Data.RowList - simple', function() {
...
+            result = rowList.findRows(conditions);
+
+            expect(result.length).toBe(1);
+            expect(result[0].c1).toBe('baz');
+        });
+
+        it('when the conditions type is object and not find rows.', function() {
+            conditions = {
+              c1: 'baz',
+              c2: 10000
```
[the remark this answers is not in the document]
Comment (seonim-ryu) [reply test/unit/js/model/data/rowList.simple.spec.js]: 룰이 동작을 안했나보네요 수정했습니다~

File: src/js/model/data/rowList.js (hunk 1 above)
Comment (seonim-ryu) [reply src/js/model/data/rowList.js]: 이거.. 문제가 있습니다...
반환되는 `foundRows` 값은 컬렉션이 아니라 배열 값이라 `toJSON`이 없습니다만..

**Pull request merged**